hyperspy/_signals/eds_tem.py

```diff
@@ -325,12 +363,33 @@ def quantification(self,
             navigation_mask = navigation_mask.data
         xray_lines = self.metadata.Sample.xray_lines
         composition = utils.stack(intensities)
-        composition.data = utils_eds.quantification_cliff_lorimer(
-            composition.data, kfactors=kfactors,
-            mask=navigation_mask) * 100.
-        composition = composition.split()
+        if method == 'CL':
+            composition.data = utils_eds.quantification_cliff_lorimer(
+                composition.data, kfactors=factors,
+                mask=navigation_mask) * 100.
+        elif method == 'zeta':
+            results = utils_eds.quantification_zeta_factor(
+                composition.data, zfactors=factors,
+                dose=self._get_dose(method))
+            composition.data = results[0] * 100.
+            mass_thickness = intensities[0].deepcopy()
+            mass_thickness.data = results[1]
+            mass_thickness.metadata.General.title = 'Mass thickness'
+        else:
```

Comment:
Oops no! I think I deleted one too many error calls :) 

> On 11 Apr 2016, at 15:16, Duncan Johnstone <notifications@github.com> wrote:
> 
> In hyperspy/_signals/eds_tem.py <https://github.com/hyperspy/hyperspy/pull/913#discussion_r59201308>:
> 
> > -            composition.data, kfactors=kfactors,
> > -            mask=navigation_mask) * 100.
> > -        composition = composition.split()
> > +        if method == 'CL':
> > +            composition.data = utils_eds.quantification_cliff_lorimer(
> > +                composition.data, kfactors=factors,
> > +                mask=navigation_mask) * 100.
> > +        elif method == 'zeta':
> > +            results = utils_eds.quantification_zeta_factor(
> > +                composition.data, zfactors=factors,
> > +                dose=self._get_dose(method))
> > +            composition.data = results[0] * 100.
> > +            mass_thickness = intensities[0].deepcopy()
> > +            mass_thickness.data = results[1]
> > +            mass_thickness.metadata.General.title = 'Mass thickness'
> > +        else:
> Doesn't this mean that if method is not 'CL' or 'zeta' for example one might set 'method=bananas' the code will give the result according to cross-sections? Is that the behaviour you intended?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub <https://github.com/hyperspy/hyperspy/pull/913/files/23ce89b24f0c3f0ca8b5e2cf424c505eb8d6be6a#r59201308>